### PR TITLE
Prep

### PR DESCRIPTION
**Pull request by jshawl**

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -1,28 +1,484 @@
...
 
+## Learning Objectives
+
+- Compare/contrast features of ES5 and ES6
+- Explain when to use var vs let vs const
+- Use default parameters and arrow functions
+- Use template literals to interpolate variables and strings
+- Compare ES6 classes w/ ES5 prototypes
+- Use import/export syntax with modules
+
```
Comment (nolds9) [inline readme.md:12]: Maybe add a LO relating to deconstruction...
- Use deconstruction to extract values from objects and arrays

File: readme.md [hunk 2]
```diff
@@ -1,28 +1,484 @@
...
+### Why now?
+
+Many plugins, frameworks and modules still use ES5, as browser support for
+the new version of the language is still limited [ahem... ie.](http://caniuse.com/#search=es6)
+
+Later today and tomorrow, we'll be working with a framework called React, and many
+React tutorials (and our lesson plans) have adopted the new syntax and features
+of ES6. Today is all about exploring those features and getting comfortable with
+the new syntax.
+
```
Comment (nolds9) [inline readme.md:34]: Are you planning on just having them use ES6 in the browser? Might be appropriate to talk about some of the tooling out there to allow you to use ES6 now anywhere with tools like Babel.

File: readme.md [hunk 3]
```diff
@@ -1,28 +1,484 @@
...
+{
+  let a = 2
+  console.log(a)
+}
+console.log(a)
+```
+
+You're more likely to see `let` declarations inside an `if` or `for` block:
+
+```js
```
Comment (nolds9) [inline readme.md:67]: might be helpful to include comments on the snippets for `es5 vs es6`

File: readme.md [hunk 4]
```diff
@@ -1,28 +1,484 @@
...
+
+Remember default parameters from Ruby, now we can do something similar in JavaScript!
+
+```js
+function hello( name = "stranger"){
+  console.log("Hello, " + name)
+}
+
+hello()
+hello("Jesse")
```
Comment (nolds9) [inline readme.md]: might be helpful to include output as a comment for these

File: readme.md [hunk 5]
```diff
@@ -1,28 +1,484 @@
...
+### Default parameters
+
+Remember default parameters from Ruby, now we can do something similar in JavaScript!
+
+```js
+function hello( name = "stranger"){
+  console.log("Hello, " + name)
+}
+
+hello()
```
Comment (nolds9) [inline readme.md]:
``` js
// "Hello, stranger"
```

File: readme.md (hunk 4 above)
Comment (nolds9) [inline readme.md]:
``` js
// "Hello, Jesse"
```

File: readme.md [hunk 6]
```diff
@@ -1,28 +1,484 @@
...
+### Destructuring
+
+Destructuring assignment makes it possible to extract data from complex data
+types (arrays and objects) into distinct variables:
+
+```js
+let [a,b] = [1,2]
+let nums = [1,2,3,4,5]
+let [first, second, ...thirdfourthfifth] = nums
+```
```
Comment (nolds9) [inline readme.md:189]: maybe include the references:

``` js
a //= 1
b //= 2
first //= 1
second //= 2
thirdfourthfifth //= [3,4,5]
```

File: readme.md [hunk 7]
```diff
@@ -1,28 +1,484 @@
...
+
+This also applies to objects:
+
+```js
+var user = {
+   id: 1,
+   name: "Bob",
+    age: 43 ,
+   profile_url:  "http://api.co/users/1"
```
Comment (nolds9) [inline readme.md:198]: spacing

File: readme.md [hunk 8]
```diff
@@ -1,28 +1,484 @@
...
+}
+
+// ES5
+function getUserInfo (user) {
+     return $.getJSON(user.profile_url)
+}
```
Comment (nolds9) [inline readme.md:204]: indendation

File: readme.md [hunk 9]
```diff
@@ -1,28 +1,484 @@
...
+  }, 1000)
+}
+
+var bob = new Person()
+
+// vs ES6
+
+function Person(){
+  this.age = 0
+  setInterval(() => {
```
Comment (nolds9) [inline readme.md:350]: something interesting to note too about Arrow functions is the return value. If on a single line - will implicitly return. i.e. `let add = (x, y) => x + y`

 if multi-line inside a block  need to explicitly return. 

``` js
let add = (x,y) => {
  return x + y
}
```

 However can fake the single line by wrapping the multi-line body in parens `()`

``` js
let add = (x,y) => (
    x + y
)
```

File: readme.md [hunk 10]
```diff
@@ -1,28 +1,484 @@
...
+```js
+function Animal(name){
+  this.name = name 
+}
+Animal.prototype.speak = function(){
+  console.log("my name is " + this.name)
+}
+
+function Dog(name){
+  this.name = name
```
Comment (nolds9) [inline readme.md:398]: Does es6 class syntax eliminate the need to define a `name` property on the `Dog` class, like we had to do with the es5 example?

``` js
function Dog (name) {
  this.name = name
}
```

File: readme.md [hunk 11]
```diff
@@ -1,28 +1,484 @@
...
+  constructor(name) {
+    this.name = name;
+  }
+  speak() {
+    console.log(this.name + ' makes a noise.');
+  }
+}
+
+class Dog extends Animal {
+  speak() {
```
Comment (nolds9) [inline readme.md:419]: no need for a constructor that takes in a name?

File: readme.md [hunk 12]
```diff
@@ -1,28 +1,484 @@
...
+
+//Import a member with a more convenient alias. This inserts shortName into the current scope.
+import {reallyReallyLongModuleMemberName as shortName} from "my-module"
+
+// Import multiple members of a module with convenient aliases.
+import {reallyReallyLongModuleMemberName as shortName, anotherLongModuleName as short} from "my-module"
+
+// Import an entire module for side effects only, without importing any bindings.
+import "my-module"
+```
```
Comment (nolds9) [inline readme.md:500]: Maybe a note about the `default` behavior of the new es6 system and what that keyword means...

File: readme.md (hunk 9 above)
Comment (jshawl) [inline readme.md:350]: COOL

File: readme.md (hunk 11 above)
Comment (jshawl) [inline readme.md:419]: nope!

**Pull request merged**